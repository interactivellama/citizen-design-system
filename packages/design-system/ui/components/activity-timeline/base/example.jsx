// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import { ButtonIcon } from 'ui/components/button-icons/base/example';
import { ActivityTimelineItem as Event } from 'ui/components/activity-timeline/event/example';
import { ActivityTimelineItem as Call } from 'ui/components/activity-timeline/call/example';
import { ActivityTimelineItem as Email } from 'ui/components/activity-timeline/email/example';
import { ActivityTimelineItem as Task } from 'ui/components/activity-timeline/task/example';
import SvgIcon from 'app_modules/ui/svg-icon';
import classNames from 'classnames';

export let TimelineIcon = props =>
  <div
    className={classNames(
      'slds-icon_container',
      props.symbol == 'log_a_call' ?
        'slds-icon-standard-log-a-call'
      : 'slds-icon-standard-' + props.symbol
    )}
    title={props.symbol == 'log_a_call' ? 'call' : props.symbol }
  >
    <SvgIcon
      className="slds-icon slds-icon--small"
      sprite="standard"
      symbol={ props.symbol }
    />
  </div>;

export let TimelineActions = props =>
  <div className="slds-timeline__actions">
    <p className="slds-timeline__date">{ props.date || 'Feb 24' }</p>
    <ButtonIcon
      className="slds-button--icon-border-filled slds-button--icon-x-small"
      symbol="down"
      assistiveText={ props.assistiveText || 'More Options' }
      aria-haspopup="true"
      title={ props.title || 'More Options' }
    />
  </div>;

export default (
  <ul className="slds-timeline">
    <li>
      <Task />
    </li>
    <li>
      <Event />
    </li>
    <li>
      <Call />
    </li>
    <li>
      <Email />
    </li>
  </ul>
);
