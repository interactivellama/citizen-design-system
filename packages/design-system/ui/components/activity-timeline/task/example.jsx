// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import MediaObject from 'ui/utilities/media-objects/index.react';
import { Checkbox } from 'ui/components/checkbox/base/example';
import { TimelineIcon, TimelineActions } from 'ui/components/activity-timeline/base/example';

export let ActivityTimelineItem = props =>
  <div className="slds-timeline__item">
    <span className="slds-assistive-text">Task</span>
    <MediaObject
      figureRight={
        <TimelineActions
          title="More Options for Task, Review proposals"
          assistiveText="More Options for Task, Review proposals"
        />
      }
    >
      <MediaObject
        className="slds-timeline__media slds-timeline__media--task"
        figureLeft={ <TimelineIcon symbol="task" /> }
        figureLeftClassName="slds-timeline__icon"
      >
        <MediaObject
          title="Review proposals for EBC deck with larger team and have marketing review this"
          figureLeft={ <Checkbox hideLabel label="mark-complete" /> }
        >
          <h3 className="slds-truncate" title="Review proposals for EBC deck with larger team and have marketing review this">
            <a href="javascript:void(0);">Review proposals for EBC deck with larger team and have marketing review this</a>
          </h3>
          <ul className="slds-list--horizontal slds-wrap">
            <li className="slds-m-right--large">
              <span className="slds-text-title">Contact:</span>
              <span className="slds-text-body--small"><a href="javascript:void(0);">Lei Chan</a></span>
            </li>
            <li className="slds-m-right--large">
              <span className="slds-text-title">Assigned to:</span>
              <span className="slds-text-body--small"><a href="javascript:void(0);">Betty Mason</a></span>
            </li>
          </ul>
        </MediaObject>
      </MediaObject>
    </MediaObject>
  </div>;

export default (
  <ActivityTimelineItem />
);
