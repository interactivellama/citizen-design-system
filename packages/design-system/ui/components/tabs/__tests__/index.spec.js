/* eslint-env jest */
import React from 'react';

import {
  TabsDefault,
  TabsWithNestedScopedTabs,
  TabsOverflow,
  TabsMedium,
  TabsLarge
} from '../base/example';

import {
  Subtab,
  Subtabs,
  SubtabList,
  SubtabPanel,
  SubtabOverflow
} from '../../tabs/sub-tabs/example';

import MobileTabs, {
  MobileTabItem,
  MobileTabList,
  MobileTabPanel,
  MobileTabButton,
  MobileTabContainer,
  MobileTabGroup,
  NestedMobileStacked,
  DeprecatedMobileTabItem,
  DeprecatedMobileTabList,
  DeprecatedMobileTabTitle,
  DeprecatedMobileTabAction,
  DeprecatedMobileStacked,
  DeprecatedAdjacentMobileStacked
} from '../mobile-stack/';

import SvgIcon from '../../../shared/svg-icon';
import { Menu, MenuList, MenuItem } from '../../menus/dropdown/example';

import createHelpers from '../../../../jest.helpers';

const { matchesMarkupAndStyle } = createHelpers(__dirname);

describe('Default Tabs', () => {
  it('renders default tabs', () => matchesMarkupAndStyle(<TabsDefault />));
  it('renders default tabs with nested scoped tabs', () =>
    matchesMarkupAndStyle(<TabsWithNestedScopedTabs />));
  it('renders default tabs with overflowing tab items', () =>
    matchesMarkupAndStyle(<TabsOverflow />));
  it('renders medium sized default tabs', () =>
    matchesMarkupAndStyle(<TabsMedium />));
  it('renders large sized default tabs', () =>
    matchesMarkupAndStyle(<TabsLarge />));
});

describe('Subtabs', () => {
  it('renders subtab with notification and dirty', () =>
    matchesMarkupAndStyle(
      <div className="slds-tabs_default slds-sub-tabs">
        <ul className="slds-tabs_default__nav" role="tablist">
          <Subtab
            title="00071938"
            active
            tabItemId="subtab-tabitem-01"
            tabPanelId="subtab-tabpanel-01"
          />
          <Subtab
            title="Chat - Customer"
            hasNotification
            itemUnsaved
            tabItemId="subtab-tabitem-02"
            tabPanelId="subtab-tabpanel-02"
            symbol="live_chat"
          />
        </ul>
        <div
          className="slds-tabs_default__content slds-show"
          id="subtab-tabpanel-01"
          role="tabpanel"
          aria-labelledby="subtab-tabitem-01"
        >
          Item One Content
        </div>
        <div
          className="slds-tabs_default__content slds-hide"
          id="subtab-tabpanel-02"
          role="tabpanel"
          aria-labelledby="subtab-tabitem-02"
        >
          Item Two Content
        </div>
      </div>
    ));

  it('renders overflowed subtab', () =>
    matchesMarkupAndStyle(
      <div
        className="demo-only"
        style={{ height: '8rem', overflow: 'visible' }}
      >
        <Subtabs>
          <SubtabList>
            <Subtab
              title="00071938"
              tabItemId="subtab-tabitem-01"
              tabPanelId="subtab-tabpanel-01"
            />
            <Subtab
              title="00071939"
              tabItemId="subtab-tabitem-02"
              tabPanelId="subtab-tabpanel-02"
            />
            <SubtabOverflow isOpen />
          </SubtabList>
          <SubtabPanel
            tabId="subtab-tabitem-01"
            id="subtab-tabpanel-01"
            isVisible
          >
            Item One Content
          </SubtabPanel>
          <SubtabPanel tabId="subtab-tabitem-02" id="subtab-tabpanel-02">
            Item Two Content
          </SubtabPanel>
        </Subtabs>
      </div>
    ));

  it('renders overflowed subtab with notification', () =>
    matchesMarkupAndStyle(
      <div
        className="demo-only"
        style={{ height: '8rem', overflow: 'visible' }}
      >
        <Subtabs>
          <SubtabList>
            <Subtab
              title="00071938"
              tabItemId="subtab-tabitem-01"
              tabPanelId="subtab-tabpanel-01"
            />
            <Subtab
              title="00071939"
              tabItemId="subtab-tabitem-02"
              tabPanelId="subtab-tabpanel-02"
            />
            <SubtabOverflow isOpen itemUnread />
          </SubtabList>
          <SubtabPanel
            tabId="subtab-tabitem-01"
            id="subtab-tabpanel-01"
            isVisible
          >
            Item One Content
          </SubtabPanel>
          <SubtabPanel tabId="subtab-tabitem-02" id="subtab-tabpanel-02">
            Item Two Content
          </SubtabPanel>
        </Subtabs>
      </div>
    ));

  it('renders dirty overflowed subtab', () =>
    matchesMarkupAndStyle(
      <div
        className="demo-only"
        style={{ height: '8rem', overflow: 'visible' }}
      >
        <Subtabs>
          <SubtabList>
            <Subtab
              title="00071938"
              tabItemId="subtab-tabitem-01"
              tabPanelId="subtab-tabpanel-01"
            />
            <Subtab
              title="00071939"
              tabItemId="subtab-tabitem-02"
              tabPanelId="subtab-tabpanel-02"
            />
            <SubtabOverflow isOpen itemUnsaved />
          </SubtabList>
          <SubtabPanel
            tabId="subtab-tabitem-01"
            id="subtab-tabpanel-01"
            isVisible
          >
            Item One Content
          </SubtabPanel>
          <SubtabPanel tabId="subtab-tabitem-02" id="subtab-tabpanel-02">
            Item Two Content
          </SubtabPanel>
        </Subtabs>
      </div>
    ));

  it('renders dirty overflowed subtab with notification', () =>
    matchesMarkupAndStyle(
      <div
        className="demo-only"
        style={{ height: '8rem', overflow: 'visible' }}
      >
        <Subtabs>
          <SubtabList>
            <Subtab
              title="00071938"
              tabItemId="subtab-tabitem-01"
              tabPanelId="subtab-tabpanel-01"
            />
            <Subtab
              title="00071939"
              tabItemId="subtab-tabitem-02"
              tabPanelId="subtab-tabpanel-02"
            />
            <SubtabOverflow isOpen itemUnsaved itemUnread />
          </SubtabList>
          <SubtabPanel
            tabId="subtab-tabitem-01"
            id="subtab-tabpanel-01"
            isVisible
          >
            Item One Content
          </SubtabPanel>
          <SubtabPanel tabId="subtab-tabitem-02" id="subtab-tabpanel-02">
            Item Two Content
          </SubtabPanel>
        </Subtabs>
      </div>
    ));
});
describe('Mobile stacked tabs', () => {
  it('renders a tab group', () => matchesMarkupAndStyle(<MobileTabGroup />));

  it('renders a tab container', () =>
    matchesMarkupAndStyle(<MobileTabContainer />));

  it('renders a tab', () => matchesMarkupAndStyle(<MobileTabItem />));

  it('renders a tab button', () =>
    matchesMarkupAndStyle(<MobileTabButton summary="List 1 Item One" />));

  it('renders a hidden tab panel', () =>
    matchesMarkupAndStyle(<MobileTabPanel title="Item One Panel" />));

  it('renders a visible tab panel', () =>
    matchesMarkupAndStyle(<MobileTabPanel title="Item One Panel" isVisible />));

  it('renders a tabset', () => matchesMarkupAndStyle(<MobileTabList />));

  it('renders a hidden tabset', () =>
    matchesMarkupAndStyle(<MobileTabList isHidden />));

  it('renders a mobile tabs demo', () => matchesMarkupAndStyle(<MobileTabs />));

  it('renders a static mobile tabs demo', () =>
    matchesMarkupAndStyle(<MobileTabs isStatic />));

  it('renders a static mobile tabs demo with a visible panel showing', () =>
    matchesMarkupAndStyle(<MobileTabs visiblePanelIdx={0} isStatic />));

  it('renders a mobile tabs demo with a visible panel showing', () =>
    matchesMarkupAndStyle(<MobileTabs visiblePanelIdx={0} />));

  it('renders nested mobile tabs', () =>
    matchesMarkupAndStyle(<NestedMobileStacked />));

  it('renders a deprecated tab item', () =>
    matchesMarkupAndStyle(<DeprecatedMobileTabItem />));

  it('renders a deprecated tab list', () =>
    matchesMarkupAndStyle(<DeprecatedMobileTabList />));

  it('renders a deprecated tab title', () =>
    matchesMarkupAndStyle(<DeprecatedMobileTabTitle />));

  it('renders a deprecated tab title action', () =>
    matchesMarkupAndStyle(
      <DeprecatedMobileTabAction summary="List 1 Item One" />
    ));

  it('renders a deprecated tab stack', () =>
    matchesMarkupAndStyle(<DeprecatedMobileStacked />));

  it('renders a deprecated adjacent tab stack', () =>
    matchesMarkupAndStyle(<DeprecatedAdjacentMobileStacked />));
});
