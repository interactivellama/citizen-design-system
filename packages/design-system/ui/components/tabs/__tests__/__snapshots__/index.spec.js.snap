// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Tabs renders default tabs 1`] = `
<div
  className="slds-tabs_default"
>
  <ul
    className="slds-tabs_default__nav"
    role="tablist"
  >
    <li
      className="slds-tabs_default__item slds-is-active"
      role="presentation"
      title="Item One"
    >
      <a
        aria-controls="tab-default-1"
        aria-selected={true}
        className="slds-tabs_default__link"
        href="javascript:void(0);"
        id="tab-default-1__item"
        role="tab"
        tabIndex={0}
      >
        Item One
      </a>
    </li>
    <li
      className="slds-tabs_default__item"
      role="presentation"
      title="Item Two"
    >
      <a
        aria-controls="tab-default-2"
        aria-selected={false}
        className="slds-tabs_default__link"
        href="javascript:void(0);"
        id="tab-default-2__item"
        role="tab"
        tabIndex={-1}
      >
        Item Two
      </a>
    </li>
    <li
      className="slds-tabs_default__item"
      role="presentation"
      title="Item Three"
    >
      <a
        aria-controls="tab-default-3"
        aria-selected={false}
        className="slds-tabs_default__link"
        href="javascript:void(0);"
        id="tab-default-3__item"
        role="tab"
        tabIndex={-1}
      >
        Item Three
      </a>
    </li>
  </ul>
  <div
    aria-labelledby="tab-default-1__item"
    className="slds-tabs_default__content slds-show"
    id="tab-default-1"
    role="tabpanel"
  >
    Item One Content
  </div>
  <div
    aria-labelledby="tab-default-2__item"
    className="slds-tabs_default__content slds-hide"
    id="tab-default-2"
    role="tabpanel"
  >
    Item Two Content
  </div>
  <div
    aria-labelledby="tab-default-3__item"
    className="slds-tabs_default__content slds-hide"
    id="tab-default-3"
    role="tabpanel"
  >
    Item Three Content
  </div>
</div>
`;

exports[`Default Tabs renders default tabs with nested scoped tabs 1`] = `
<div
  className="slds-tabs_default"
>
  <ul
    className="slds-tabs_default__nav"
    role="tablist"
  >
    <li
      className="slds-tabs_default__item slds-is-active"
      role="presentation"
      title="Item One"
    >
      <a
        aria-controls="tab-default-1"
        aria-selected={true}
        className="slds-tabs_default__link"
        href="javascript:void(0);"
        id="tab-default-1__item"
        role="tab"
        tabIndex={0}
      >
        Item One
      </a>
    </li>
    <li
      className="slds-tabs_default__item"
      role="presentation"
      title="Item Two"
    >
      <a
        aria-controls="tab-default-2"
        aria-selected={false}
        className="slds-tabs_default__link"
        href="javascript:void(0);"
        id="tab-default-2__item"
        role="tab"
        tabIndex={-1}
      >
        Item Two
      </a>
    </li>
    <li
      className="slds-tabs_default__item"
      role="presentation"
      title="Item Three"
    >
      <a
        aria-controls="tab-default-3"
        aria-selected={false}
        className="slds-tabs_default__link"
        href="javascript:void(0);"
        id="tab-default-3__item"
        role="tab"
        tabIndex={-1}
      >
        Item Three
      </a>
    </li>
  </ul>
  <div
    aria-labelledby="tab-default-1__item"
    className="slds-tabs_default__content slds-show"
    id="tab-default-1"
    role="tabpanel"
  >
    <div
      className="slds-tabs_scoped"
    >
      <ul
        className="slds-tabs_scoped__nav"
        role="tablist"
      >
        <li
          className="slds-tabs_scoped__item slds-is-active"
          role="presentation"
          title="Item One"
        >
          <a
            aria-controls="tab-scoped-1"
            aria-selected={true}
            className="slds-tabs_scoped__link"
            href="javascript:void(0);"
            id="tab-scoped-1__item"
            role="tab"
            tabIndex={0}
          >
            Item One
          </a>
        </li>
        <li
          className="slds-tabs_scoped__item"
          role="presentation"
          title="Item Two"
        >
          <a
            aria-controls="tab-scoped-2"
            aria-selected={false}
            className="slds-tabs_scoped__link"
            href="javascript:void(0);"
            id="tab-scoped-2__item"
            role="tab"
            tabIndex={-1}
          >
            Item Two
          </a>
        </li>
        <li
          className="slds-tabs_scoped__item"
          role="presentation"
          title="Item Three"
        >
          <a
            aria-controls="tab-scoped-3"
            aria-selected={false}
            className="slds-tabs_scoped__link"
            href="javascript:void(0);"
            id="tab-scoped-3__item"
            role="tab"
            tabIndex={-1}
          >
            Item Three
          </a>
        </li>
      </ul>
      <div
        aria-labelledby="tab-scoped-1__item"
        className="slds-tabs_scoped__content slds-show"
        id="tab-scoped-1"
        role="tabpanel"
      >
        Item One Content
      </div>
      <div
        aria-labelledby="tab-scoped-2__item"
        className="slds-tabs_scoped__content slds-hide"
        id="tab-scoped-2"
        role="tabpanel"
      >
        Item Two Content
      </div>
      <div
        aria-labelledby="tab-scoped-3__item"
        className="slds-tabs_scoped__content slds-hide"
        id="tab-scoped-3"
        role="tabpanel"
      >
        Item Three Content
      </div>
    </div>
  </div>
  <div
    aria-labelledby="tab-default-2__item"
    className="slds-tabs_default__content slds-hide"
    id="tab-default-2"
    role="tabpanel"
  >
    Item Two Content
  </div>
  <div
    aria-labelledby="tab-default-3__item"
    className="slds-tabs_default__content slds-hide"
    id="tab-default-3"
    role="tabpanel"
  >
    Item Three Content
  </div>
</div>
`;

exports[`Default Tabs renders default tabs with overflowing tab items 1`] = `
<div
  className="demo-only"
  style={
    Object {
      "height": "12rem",
    }
  }
>
  <div
    className="slds-tabs_default"
  >
    <ul
      className="slds-tabs_default__nav"
      role="tablist"
    >
      <li
        className="slds-tabs_default__item slds-is-active"
        role="presentation"
        title="Item One"
      >
        <a
          aria-controls="tab-default-1"
          aria-selected={true}
          className="slds-tabs_default__link"
          href="javascript:void(0);"
          id="tab-default-1__item"
          role="tab"
          tabIndex={0}
        >
          Item One
        </a>
      </li>
      <li
        className="slds-tabs_default__item"
        role="presentation"
        title="Item Two"
      >
        <a
          aria-controls="tab-default-2"
          aria-selected={false}
          className="slds-tabs_default__link"
          href="javascript:void(0);"
          id="tab-default-2__item"
          role="tab"
          tabIndex={-1}
        >
          Item Two
        </a>
      </li>
      <li
        className="slds-tabs_default__item"
        role="presentation"
        title="Item Three"
      >
        <a
          aria-controls="tab-default-3"
          aria-selected={false}
          className="slds-tabs_default__link"
          href="javascript:void(0);"
          id="tab-default-3__item"
          role="tab"
          tabIndex={-1}
        >
          Item Three
        </a>
      </li>
      <li
        className="slds-tabs_default__item"
        role="presentation"
        title="Item Four"
      >
        <a
          aria-controls="tab-default-4"
          aria-selected={false}
          className="slds-tabs_default__link"
          href="javascript:void(0);"
          id="tab-default-4__item"
          role="tab"
          tabIndex={-1}
        >
          Item Four
        </a>
      </li>
      <li
        className="slds-tabs_default__item"
        role="presentation"
        title="Item Five"
      >
        <a
          aria-controls="tab-default-5"
          aria-selected={false}
          className="slds-tabs_default__link"
          href="javascript:void(0);"
          id="tab-default-5__item"
          role="tab"
          tabIndex={-1}
        >
          Item Five
        </a>
      </li>
      <li
        className="slds-tabs_default__item"
        role="presentation"
        title="Item Six"
      >
        <a
          aria-controls="tab-default-6"
          aria-selected={false}
          className="slds-tabs_default__link"
          href="javascript:void(0);"
          id="tab-default-6__item"
          role="tab"
          tabIndex={-1}
        >
          Item Six
        </a>
      </li>
      <li
        className="slds-tabs_default__item"
        role="presentation"
        title="Item Seven"
      >
        <a
          aria-controls="tab-default-7"
          aria-selected={false}
          className="slds-tabs_default__link"
          href="javascript:void(0);"
          id="tab-default-7__item"
          role="tab"
          tabIndex={-1}
        >
          Item Seven
        </a>
      </li>
      <li
        className="slds-tabs_default__item slds-tabs_default__overflow-button"
        role="presentation"
        title="More Tabs"
      >
        <div
          className="slds-dropdown-trigger slds-dropdown-trigger_click slds-is-open"
        >
          <button
            aria-haspopup={true}
            className="slds-button"
            disabled={undefined}
          >
            More
            <svg
              aria-hidden={true}
              className="slds-button__icon slds-button__icon_x-small slds-button__icon_right"
            >
              <use
                xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#down"
              />
            </svg>
          </button>
          <div
            className="slds-dropdown slds-dropdown_right"
            style={undefined}
          >
            <ul
              aria-label={undefined}
              className="slds-dropdown__list slds-dropdown_length-with-icon-10"
              role="menu"
            >
              <li
                className="slds-dropdown__item"
                role="presentation"
              >
                <a
                  aria-checked={undefined}
                  href="javascript:void(0);"
                  role="menuitem"
                  tabIndex="-1"
                >
                  <span
                    className="slds-truncate"
                    title="Accounts"
                  >
                    <span
                      className="slds-media slds-media_center"
                    >
                      <span
                        className="slds-media__figure"
                      >
                        <span
                          className="slds-icon_container slds-icon-standard-account"
                          title="Description of icon when needed"
                        >
                          <svg
                            aria-hidden={true}
                            className="slds-icon slds-icon_small"
                          >
                            <use
                              xlinkHref="/assets/icons/standard-sprite/svg/symbols.svg#account"
                            />
                          </svg>
                          <span
                            className="slds-assistive-text"
                          >
                            Description of icon when needed
                          </span>
                        </span>
                      </span>
                      <span
                        className="slds-media__body"
                      >
                        Accounts
                      </span>
                    </span>
                  </span>
                </a>
              </li>
              <li
                className="slds-dropdown__item"
                role="presentation"
              >
                <a
                  aria-checked={undefined}
                  href="javascript:void(0);"
                  role="menuitem"
                  tabIndex="-1"
                >
                  <span
                    className="slds-truncate"
                    title="Approvals"
                  >
                    <span
                      className="slds-media slds-media_center"
                    >
                      <span
                        className="slds-media__figure"
                      >
                        <span
                          className="slds-icon_container slds-icon-standard-approval"
                          title="Description of icon when needed"
                        >
                          <svg
                            aria-hidden={true}
                            className="slds-icon slds-icon_small"
                          >
                            <use
                              xlinkHref="/assets/icons/standard-sprite/svg/symbols.svg#approval"
                            />
                          </svg>
                          <span
                            className="slds-assistive-text"
                          >
                            Description of icon when needed
                          </span>
                        </span>
                      </span>
                      <span
                        className="slds-media__body"
                      >
                        Approvals
                      </span>
                    </span>
                  </span>
                </a>
              </li>
              <li
                className="slds-dropdown__item"
                role="presentation"
              >
                <a
                  aria-checked={undefined}
                  href="javascript:void(0);"
                  role="menuitem"
                  tabIndex="-1"
                >
                  <span
                    className="slds-truncate"
                    title="Lead"
                  >
                    <span
                      className="slds-media slds-media_center"
                    >
                      <span
                        className="slds-media__figure"
                      >
                        <span
                          className="slds-icon_container slds-icon-standard-lead"
                          title="Description of icon when needed"
                        >
                          <svg
                            aria-hidden={true}
                            className="slds-icon slds-icon_small"
                          >
                            <use
                              xlinkHref="/assets/icons/standard-sprite/svg/symbols.svg#lead"
                            />
                          </svg>
                          <span
                            className="slds-assistive-text"
                          >
                            Description of icon when needed
                          </span>
                        </span>
                      </span>
                      <span
                        className="slds-media__body"
                      >
                        Leads
                      </span>
                    </span>
                  </span>
                </a>
              </li>
            </ul>
          </div>
        </div>
      </li>
    </ul>
    <div
      aria-labelledby="tab-default-1__item"
      className="slds-tabs_default__content slds-show"
      id="tab-default-1"
      role="tabpanel"
    >
      Item One Content
    </div>
    <div
      aria-labelledby="tab-default-2__item"
      className="slds-tabs_default__content slds-hide"
      id="tab-default-2"
      role="tabpanel"
    >
      Item Two Content
    </div>
    <div
      aria-labelledby="tab-default-3__item"
      className="slds-tabs_default__content slds-hide"
      id="tab-default-3"
      role="tabpanel"
    >
      Item Three Content
    </div>
    <div
      aria-labelledby="tab-default-4__item"
      className="slds-tabs_default__content slds-hide"
      id="tab-default-4"
      role="tabpanel"
    >
      <h2>
        Item Four Content
      </h2>
    </div>
    <div
      aria-labelledby="tab-default-5__item"
      className="slds-tabs_default__content slds-hide"
      id="tab-default-5"
      role="tabpanel"
    >
      <h2>
        Item Five Content
      </h2>
    </div>
    <div
      aria-labelledby="tab-default-6__item"
      className="slds-tabs_default__content slds-hide"
      id="tab-default-6"
      role="tabpanel"
    >
      <h2>
        Item Six Content
      </h2>
    </div>
    <div
      aria-labelledby="tab-default-7__item"
      className="slds-tabs_default__content slds-hide"
      id="tab-default-7"
      role="tabpanel"
    >
      <h2>
        Item Seven Content
      </h2>
    </div>
  </div>
</div>
`;

exports[`Default Tabs renders large sized default tabs 1`] = `
<div
  className="slds-tabs_default slds-tabs_large"
>
  <ul
    className="slds-tabs_default__nav"
    role="tablist"
  >
    <li
      className="slds-tabs_default__item slds-is-active"
      role="presentation"
      title="Item One"
    >
      <a
        aria-controls="tab-default-1"
        aria-selected={true}
        className="slds-tabs_default__link"
        href="javascript:void(0);"
        id="tab-default-1__item"
        role="tab"
        tabIndex={0}
      >
        Item One
      </a>
    </li>
    <li
      className="slds-tabs_default__item"
      role="presentation"
      title="Item Two"
    >
      <a
        aria-controls="tab-default-2"
        aria-selected={false}
        className="slds-tabs_default__link"
        href="javascript:void(0);"
        id="tab-default-2__item"
        role="tab"
        tabIndex={-1}
      >
        Item Two
      </a>
    </li>
    <li
      className="slds-tabs_default__item"
      role="presentation"
      title="Item Three"
    >
      <a
        aria-controls="tab-default-3"
        aria-selected={false}
        className="slds-tabs_default__link"
        href="javascript:void(0);"
        id="tab-default-3__item"
        role="tab"
        tabIndex={-1}
      >
        Item Three
      </a>
    </li>
  </ul>
  <div
    aria-labelledby="tab-default-1__item"
    className="slds-tabs_default__content slds-show"
    id="tab-default-1"
    role="tabpanel"
  >
    Item One Content
  </div>
  <div
    aria-labelledby="tab-default-2__item"
    className="slds-tabs_default__content slds-hide"
    id="tab-default-2"
    role="tabpanel"
  >
    Item Two Content
  </div>
  <div
    aria-labelledby="tab-default-3__item"
    className="slds-tabs_default__content slds-hide"
    id="tab-default-3"
    role="tabpanel"
  >
    Item Three Content
  </div>
</div>
`;

exports[`Default Tabs renders medium sized default tabs 1`] = `
<div
  className="slds-tabs_default slds-tabs_medium"
>
  <ul
    className="slds-tabs_default__nav"
    role="tablist"
  >
    <li
      className="slds-tabs_default__item slds-is-active"
      role="presentation"
      title="Item One"
    >
      <a
        aria-controls="tab-default-1"
        aria-selected={true}
        className="slds-tabs_default__link"
        href="javascript:void(0);"
        id="tab-default-1__item"
        role="tab"
        tabIndex={0}
      >
        Item One
      </a>
    </li>
    <li
      className="slds-tabs_default__item"
      role="presentation"
      title="Item Two"
    >
      <a
        aria-controls="tab-default-2"
        aria-selected={false}
        className="slds-tabs_default__link"
        href="javascript:void(0);"
        id="tab-default-2__item"
        role="tab"
        tabIndex={-1}
      >
        Item Two
      </a>
    </li>
    <li
      className="slds-tabs_default__item"
      role="presentation"
      title="Item Three"
    >
      <a
        aria-controls="tab-default-3"
        aria-selected={false}
        className="slds-tabs_default__link"
        href="javascript:void(0);"
        id="tab-default-3__item"
        role="tab"
        tabIndex={-1}
      >
        Item Three
      </a>
    </li>
  </ul>
  <div
    aria-labelledby="tab-default-1__item"
    className="slds-tabs_default__content slds-show"
    id="tab-default-1"
    role="tabpanel"
  >
    Item One Content
  </div>
  <div
    aria-labelledby="tab-default-2__item"
    className="slds-tabs_default__content slds-hide"
    id="tab-default-2"
    role="tabpanel"
  >
    Item Two Content
  </div>
  <div
    aria-labelledby="tab-default-3__item"
    className="slds-tabs_default__content slds-hide"
    id="tab-default-3"
    role="tabpanel"
  >
    Item Three Content
  </div>
</div>
`;

exports[`Subtabs renders subtab with notification and dirty 1`] = `
<div
  className="slds-tabs_default slds-sub-tabs"
>
  <ul
    className="slds-tabs_default__nav"
    role="tablist"
  >
    <li
      className="slds-tabs_default__item slds-sub-tabs__item slds-grid slds-grid_vertical-align-center slds-active"
      role="presentation"
    >
      <a
        aria-controls="subtab-tabpanel-01"
        aria-selected="true"
        className="slds-tabs_default__link slds-p-horizontal_xx-small"
        href="javascript:void(0);"
        id="subtab-tabitem-01"
        role="tab"
        tabIndex="0"
        title="00071938"
      >
        <span
          className="slds-indicator-container"
        />
        <div
          className="slds-icon_container"
          title="Case"
        >
          <svg
            aria-hidden={true}
            className="slds-icon slds-icon_small slds-icon-text-default"
          >
            <use
              xlinkHref="/assets/icons/standard-sprite/svg/symbols.svg#case"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Case
            :
          </span>
        </div>
        <span
          className="slds-truncate"
          title="00071938"
        >
          00071938
        </span>
      </a>
      <div
        className="slds-col_bump-left slds-p-left_none slds-p-right_none"
      >
        <button
          aria-haspopup={undefined}
          className="slds-button slds-button_icon slds-button_icon-container slds-button_icon-x-small"
          id={undefined}
          tabIndex="0"
          title="Close 00071938"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Close 00071938
          </span>
        </button>
      </div>
    </li>
    <li
      className="slds-tabs_default__item slds-sub-tabs__item slds-grid slds-grid_vertical-align-center slds-has-notification slds-is-unsaved"
      role="presentation"
    >
      <a
        aria-controls="subtab-tabpanel-02"
        aria-selected="false"
        className="slds-tabs_default__link slds-p-horizontal_xx-small"
        href="javascript:void(0);"
        id="subtab-tabitem-02"
        role="tab"
        tabIndex="-1"
        title="Chat - Customer"
      >
        <span
          className="slds-indicator-container"
        >
          <abbr
            aria-label="Tab Not Saved"
            className="slds-indicator_unsaved"
            title="Tab Not Saved"
          >
            *
          </abbr>
          <span
            aria-label="New Activity"
            className="slds-indicator_unread"
            role="alert"
            title="New Activity"
          >
            <span
              className="slds-assistive-text"
            >
              New activity in Tab: Chat - Customer
            </span>
          </span>
        </span>
        <div
          className="slds-icon_container"
          title="Live Chat"
        >
          <svg
            aria-hidden={true}
            className="slds-icon slds-icon_small slds-icon-text-default"
          >
            <use
              xlinkHref="/assets/icons/standard-sprite/svg/symbols.svg#live_chat"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Live Chat
            :
          </span>
        </div>
        <span
          className="slds-truncate"
          title="Chat - Customer"
        >
          Chat - Customer
        </span>
      </a>
      <div
        className="slds-col_bump-left slds-p-left_none slds-p-right_none"
      >
        <button
          aria-haspopup={undefined}
          className="slds-button slds-button_icon slds-button_icon-container slds-button_icon-x-small"
          id={undefined}
          tabIndex="-1"
          title="Close Chat - Customer"
        >
          <svg
            aria-hidden={true}
            className="slds-button__icon"
          >
            <use
              xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#close"
            />
          </svg>
          <span
            className="slds-assistive-text"
          >
            Close Chat - Customer
          </span>
        </button>
      </div>
    </li>
  </ul>
  <div
    aria-labelledby="subtab-tabitem-01"
    className="slds-tabs_default__content slds-show"
    id="subtab-tabpanel-01"
    role="tabpanel"
  >
    Item One Content
  </div>
  <div
    aria-labelledby="subtab-tabitem-02"
    className="slds-tabs_default__content slds-hide"
    id="subtab-tabpanel-02"
    role="tabpanel"
  >
    Item Two Content
  </div>
</div>
`;
